File: Luogu/JCA/JC2/P_1200_USACO_1_1_你的飞碟在这儿_Your_Ride_Is_Here.py
s = input()
t = input()
MOD = 47
x = y = 1
for i, ch in enumerate(s):
    x = x * (ord(ch) - ord('A') + 1) % MOD
for i, ch in enumerate(t):
    y = y * (ord(ch) - ord('A') + 1) % MOD

print("GO" if x % MOD == y % MOD else "STAY")

File: Luogu/Course/JCA/JC2/P_1200_USACO_1_1_你的飞碟在这儿_Your_Ride_Is_Here.cpp
#include <bits/stdc++.h>
using namespace std;
const int MOD = 47;
#define endl '\n'

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    string s, t;
    cin >> s >> t;
    int x = 1, y = 1;
    for (auto ch : s) x = x * (ch - 'A' + 1) % MOD;
    for (auto ch : t) y = y * (ch - 'A' + 1) % MOD;
    cout << (x == y ? "GO" : "STAY") << endl;
    return 0;
}